backend/server/routers/collection.py

```diff
@@ -22,130 +22,109 @@
-        client = await get_client()
-        created_collection = await client.acreate_collection(
-            collection=CreateCollection(
-                name=collection.name,
-                description=collection.description,
-                embedder_config=collection.embedder_config,
-            )
+    logger.info(f"Creating collection {collection.name}...")
+    metadata_store_client = await get_client()
+    created_collection = await metadata_store_client.acreate_collection(
+        collection=CreateCollection(
+            name=collection.name,
+            description=collection.description,
+            embedder_config=collection.embedder_config,
         )
-        logger.info(f"Creating collection {collection.name} on vector db...")
-        VECTOR_STORE_CLIENT.create_collection(
-            collection_name=collection.name,
-            embeddings=model_gateway.get_embedder_from_model_config(
-                model_name=collection.embedder_config.name
-            ),
-        )
-        logger.info(f"Created collection... {created_collection}")
-
-        if collection.associated_data_sources:
-            for data_source in collection.associated_data_sources:
-                await client.aassociate_data_source_with_collection(
-                    collection_name=created_collection.name,
-                    data_source_association=AssociateDataSourceWithCollection(
-                        data_source_fqn=data_source.data_source_fqn,
-                        parser_config=data_source.parser_config,
-                    ),
-                )
-            created_collection = await client.aget_collection_by_name(
-                collection_name=created_collection.name
+    )
+    logger.info(f"Creating collection {collection.name} on vector db...")
+    print("collection.embedder_config.name", collection.embedder_config.name)
+    print(
+        "model_gateway.get_embedder_from_model_config(collection.embedder_config.name)",
+        model_gateway.get_embedder_from_model_config(collection.embedder_config.name),
+    )
+    VECTOR_STORE_CLIENT.create_collection(
+        collection_name=collection.name,
+        embeddings=model_gateway.get_embedder_from_model_config(
+            model_name=collection.embedder_config.name
+        ),
+    )
+    logger.info(f"Created collection... {created_collection}")
+
+    if collection.associated_data_sources:
+        data_source_associations = [
+            AssociateDataSourceWithCollection(
+                data_source_fqn=data_source.data_source_fqn,
+                parser_config=data_source.parser_config,
+            )
+            for data_source in collection.associated_data_sources
+        ]
+
+        created_collection = (
+            await metadata_store_client.aassociate_data_sources_with_collection(
+                collection_name=created_collection.name,
+                data_source_associations=data_source_associations,
```

Comment:
Same goes here if we have a join table, we can just insert into the join table instead of updating collections

Reply:
https://github.com/truefoundry/cognita/pull/361#discussion_r1792465362